### PR TITLE
Refactor cronsjobs

### PR DESCRIPTION
**Pull request by garciagenrique**
#124 Needs to be accepted first.

Refactoring cron jobs : 
 - ~Cron scripts are part of the infrastructure, so moving them inside the `infra/scripts` subdir~
 - All cron ops containers depends from the same base layer
 - ~Updating paths inside cron scripts and cronjob k8s manifests <-- TODO: make cron scripts executables by adding a directory into `$PATH` env var so that we can generalise manifests~ scripts are copied into the container into the `/home` directory instead to be accessed through a `git clone`. Thus, only the scripts are inside the container, instead of the full repository. 
 - Update workflows CI scripts and dockerfiles

### CODE REVIEW

File: .github/workflows/docker-iam-rucio-sync.yml [hunk 1]
```diff
@@ -13,10 +8,11 @@ on:
       - containers/iam-rucio-sync/**
```
Comment (goseind) [inline .github/workflows/docker-iam-rucio-sync.yml:8]: I suggest we build on push to the `main` branch and add a condition for the tag (see below).

File: containers/base-layer-ops/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# VRE base layer ops container
+
+Common base container for all VRE operations.
+
+It contains:
+- A fixed version of rucio-server
+ - Useful basic tools (including latests kubectl)
+ - CERN CA certs installed from CERN maintained mirror
+ - ESCAPE VOMS and its setup
+ - EFI trust anchos
```
Comment (goseind) [inline containers/base-layer-ops/README.md:10]: indent in the first line and spelling: anchos --> anchors

File: containers/iam-rucio-sync/Dockerfile [hunk 3]
```diff
@@ -1,50 +1,13 @@
...
-yum clean all && \
-rm -rf /var/cache/yum
-
-# install useful tools
-RUN yum -y install git htop voms-clients-cpp
+ARG BASETAG=latest
+ 
+# To be changed to specific version+hsa
+FROM ghcr.io/vre-hub/vre-base-layer-ops:${BASETAG}
+LABEL maintainer="VRE Team @ CERN"
```
Comment (goseind) [inline containers/iam-rucio-sync/Dockerfile:5]: I would add these labels to the base image then.

File: containers/iam-rucio-sync/Dockerfile [hunk 4]
```diff
@@ -1,50 +1,13 @@
...
 
 WORKDIR /home/user/
 
-# # clone repos
+# # clone repo
```
Comment (goseind) [inline containers/iam-rucio-sync/Dockerfile:9]: one `#`

File: containers/iam-rucio-sync/Dockerfile [hunk 5]
```diff
@@ -1,50 +1,13 @@
...
-# ESCAPE VOMS setup
-RUN mkdir -p /etc/vomses \
-&& wget https://indigo-iam.github.io/escape-docs/voms-config/voms-escape.cloud.cnaf.infn.it.vomses -O /etc/vomses/voms-escape.cloud.cnaf.infn.it.vomses
-RUN mkdir -p /etc/grid-security/vomsdir/escape \
-&& wget https://indigo-iam.github.io/escape-docs/voms-config/voms-escape.cloud.cnaf.infn.it.lsc -O /etc/grid-security/vomsdir/escape/voms-escape.cloud.cnaf.infn.it.lsc
-
-# Install latest kubectl
-RUN curl -o /usr/bin/kubectl -L https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl
-RUN chmod +x /usr/bin/kubectl
+RUN pip install -r /home/user/vre/infrastructure/scripts/cron_scripts/iam-rucio-sync/requirements.txt
```
Comment (goseind) [inline containers/iam-rucio-sync/Dockerfile:11]: This is a local path and should be a relative path in the repo.

File: containers/iam-rucio-sync/Dockerfile [hunk 6]
```diff
@@ -1,50 +1,13 @@
...
+ 
+# To be changed to specific version+hsa
+FROM ghcr.io/vre-hub/vre-base-layer-ops:${BASETAG}
+LABEL maintainer="VRE Team @ CERN"
 
 WORKDIR /home/user/
 
-# # clone repos
+# # clone repo
 RUN git clone https://github.com/vre-hub/vre.git
```
Comment (goseind) [inline containers/iam-rucio-sync/Dockerfile:10]: Maybe this should also be versioned, as the files will change - what do you think?

File: containers/rucio-noise/Dockerfile [hunk 7]
```diff
@@ -1,48 +1,13 @@
...
-# cleanup yum cache
-RUN yum upgrade -y \
-    && yum clean all \
-    && rm -rf /var/cache/yum
-
-# install useful tools
-RUN yum -y install git htop wget voms-clients-cpp
+# To be changed to specific version+hsa
+FROM ghcr.io/vre-hub/vre-base-layer-ops:${BASETAG}
+LABEL maintainer="VRE Team @ CERN"
```
Comment (goseind) [inline containers/rucio-noise/Dockerfile:5]: I would add these labels to the base image then.

File: infrastructure/scripts/cron_scripts/rucio-noise/produce_noise.sh [hunk 8]
```diff
@@ -8,7 +8,7 @@ while read line
 do
     rses+=($line) 
 
-done < /home/user/vre/containers/rucio-noise/rses.txt
+done < /home/user/vre/infrastructure/scripts/cron_scripts/rucio-noise/rses.txt
```
Comment (goseind) [inline infrastructure/scripts/cron_scripts/rucio-noise/produce_noise.sh:11]: This is a local path and should be a relative path in the repo.

File: .github/workflows/docker-iam-rucio-sync.yml [hunk 9]
```diff
@@ -62,6 +45,14 @@ jobs:
         working-directory: containers/iam-rucio-sync
         run: |
           docker build . --tag  ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:latest
-          docker run ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:latest
           docker push  ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:latest
+          export sha=${{ github.sha }} 
+          export short_sha=${sha:0:7}
```
[the remark this answers is not in the document]
Comment (garciagenrique) [reply .github/workflows/docker-iam-rucio-sync.yml:50]: Could you point me out the env var pointing to the short sha, please ? I couldn't find it. 
About the `version.txt` I don't see any advantage of having a file instead of fetching the version directly from the repository tag. But in general, yes I agree with the approach of the tagging, of course

File: infrastructure/cluster/flux/rucio-cronjobs.yaml [hunk 10]
```diff
@@ -79,7 +79,7 @@ spec:
               command:
               - /bin/sh
               - -c
-              - date; pwd; cd /opt/rucio/etc/; ls; echo Hello from rucio-noise container; export RUCIO_CONFIG=/opt/rucio/etc/rucio-sso.cfg; echo Exported config; rucio -vvv whoami; FILE_SIZE=10M /bin/bash /home/user/vre/containers/rucio-noise/produce_noise.sh; echo 'DB cleanup Done!'
+              - date; pwd; cd /opt/rucio/etc/; ls; echo Hello from rucio-noise container; export RUCIO_CONFIG=/opt/rucio/etc/rucio-sso.cfg; echo Exported config; rucio -vvv whoami; export FSIZE=10M; FILE_SIZE=${FSIZE} /bin/bash /home/user/vre/infrastructure/scripts/cron_scripts/rucio-noise/produce_noise.sh; echo 'Rucio noise ${FSIZE} Done!'
```
[the remark this answers is not in the document]
Comment (garciagenrique) [reply infrastructure/cluster/flux/rucio-cronjobs.yaml:82]: I didn't fully get it Domenic,
If we don't install the package (what I would completely support), the only option that exists is cloning the repo. If we do so, we'd need to move directly to the relative path (locally inside the container), so I don't understand what you are mentioning. 
Also, if we are keeping the scripts that are run within a container in the same folder that builds the container, where should we put it ?

File: .github/workflows/docker-iam-rucio-sync.yml (hunk 9 above)
[the remark this answers is not in the document]
Comment (goseind) [reply .github/workflows/docker-iam-rucio-sync.yml:50]: I was mistaken, there is no short SHA.
You're right let's work with the automated SHA and try the repo versioning you suggested (ref https://github.com/vre-hub/project/issues/33).

File: infrastructure/cluster/flux/rucio-cronjobs.yaml (hunk 10 above)
[the remark this answers is not in the document]
Comment (goseind) [reply infrastructure/cluster/flux/rucio-cronjobs.yaml:82]: The part `/home/user/vre/` is your local path no? I should be relative to the `Dockerfile`?

[the remark this answers is not in the document]
Comment (garciagenrique) [reply infrastructure/cluster/flux/rucio-cronjobs.yaml:82]: When building the image, we are not installing any package, just cloning the `vre` repo. Later, when the cronjob will be executed, and because the repository was cloned and NOT installed, the script will be found in the dir relative to where the repo was clone. You are indeed running the script INSIDE the container, that it's being executed. Thus the `dockerfile` has nothing to do in this case.

In other words, if ever the repo was cloned in a different location than `/home/user/`, to be able to run the script you'd need to point the new location.

File: .github/workflows/docker-iam-rucio-sync.yml (hunk 9 above)
[the remark this answers is not in the document]
Comment (garciagenrique) [reply .github/workflows/docker-iam-rucio-sync.yml:50]: for info, I will be using `git describe` that contains the version, the number of commits from a last tagged version, plus the short sha, so basically just useful info.

**Comment by garciagenrique** [comment]
Ready to be review. @egazzarr if you can have a look at some point, it will be amazing.
 - There are still few things that I would like to fully understand; I don't think we need all the certs inside budle-ca.pem. But to be investigated in the future
